07-performance.Rmd

```diff
@@ -4,16 +4,24 @@ knit: "bookdown::preview_chapter"
 
 # Efficient optimization {#performance}
 
-[Donald Knuth](https://en.wikiquote.org/wiki/Donald_Knuth) is a famous American
-computer scientist who developed a number of the key algorithms that we use today.
-On the subject of optimisation he give this advice.
+[Donald Knuth](https://en.wikiquote.org/wiki/Donald_Knuth) is a legendary American
+computer scientist who developed a number of the key algorithms that we use today (see for 
+example `?Random`).
+On the subject of optimization he give this advice.
 
 > The real problem is that programmers have spent far too much time worrying about
 efficiency in the wrong places and at the wrong times; premature optimization is the
 root of all evil (or at least most of it) in programming.
 
 Knuth's point is that it is easy to undertake code optimisation inefficiently.
-For this reason this chapter is covered towards the latter half of the book. The previous chapters deliberately focussed on concepts, packages and functions
+When developing code the speed bottleneck is fluid and moves. This means that
```

Comment:
Suggest replacing this line with:

```
When developing code, the causes of inefficiencies may shift so that what originally caused slowness at the beginning of your work may not be relevant at a later stage. This means that
```